flycheck.el

```diff
@@ -3610,6 +3650,55 @@ list."
       ;; highlighting
       (flycheck-error-list-highlight-errors preserve-pos))))
 
+(defvar flycheck-error-list-current-filter nil
+  "The minimum level of errors to display in the error list.
+
+If set to an error level, only displays errors whose error level
+is at least as severe as this one in the error list.  If nil,
+display all errors.")
+
+(defun flycheck-error-list-mode-line-filter-indicator ()
+  "Create a string representing the current error list filter."
+  (if flycheck-error-list-current-filter
+      (format " [>= %s]" (symbol-name flycheck-error-list-current-filter))
+    ""))
+
+(defun flycheck-error-list-query-level (prompt)
+  "Reads an error level from the user.
+
+Only offers level for which errors currently exist."
+  (-when-let* ((error-levels (mapcar #'flycheck-error-level
+                                     (flycheck-error-list-current-errors)))
+               (uniq-error-levels (-uniq error-levels))
```

Comment:
I've edited the code to include the default levels (`'error`, `'warning`,  and `'info`). We could display all defined levels if we kept track of them, but do we want to list unrelated levels too? Maybe package `a` and package `b` define different levels, and it's not clear that we want to offer both options at the same time; I'll let you judge!